apps/prairielearn/src/sync/course-db.ts

```diff
@@ -511,7 +511,7 @@ export async function loadInfoFile<T extends { uuid: string }>({
 }: {
   coursePath: string;
   filePath: string;
-  schema?: JSONSchemaType<T>;
+  schema?: Schema;
```

Comment:
Presumably this was changed because of https://github.com/microsoft/TypeScript/issues/32063? AFAICT this negates your proposed benefits:

> Using these imports allows typing to be retrieved in compilation time, and also allows types to be compared against ajv Schema objects.

`Schema` is a very loose type:

```ts
interface _SchemaObject {
    id?: string;
    $id?: string;
    $schema?: string;
    [x: string]: any;
}
export interface SchemaObject extends _SchemaObject {
    id?: string;
    $id?: string;
    $schema?: string;
    $async?: false;
    [x: string]: any;
}
export interface AsyncSchema extends _SchemaObject {
    $async: true;
}
export type AnySchemaObject = SchemaObject | AsyncSchema;
export type Schema = SchemaObject | boolean;
```

Given this, are there in fact any benefits to this change?

Reply:
> Presumably this was changed because of [microsoft/TypeScript#32063](https://github.com/microsoft/TypeScript/issues/32063)?

Indeed I had issues trying to match the types, and the issue you mentioned makes sense (I hadn't read it, but it matches what I found).

> AFAICT this negates your proposed benefits:

> Given this, are there in fact any benefits to this change?

You are right, the benefits are minimal. I'll close this PR, and come back to this if there is any movement that warrants this additional check.